sklearn/cluster/tests/test_hierarchical.py

```diff
@@ -44,6 +44,9 @@ def test_linkage_misc():
     assert_raises(ValueError, linkage_tree, X, linkage='foo')
     assert_raises(ValueError, linkage_tree, X, connectivity=np.ones((4, 4)))
 
+    # Test for warning of deprecation of n_components in linkage_tree
+    children, n_nodes, n_leaves, parent = assert_warns(UserWarning, linkage_tree, n_components=10)
```

Comment:
Put it in a separate test function, please. You should probably assert that output is identical to without the n_components.